### PR TITLE
RFE-5013: RTE: more e2e tests

### PR DESCRIPTION
**Pull request by ffromani**
add e2e test to cover invalid tolerations in the spec - NROP will act as passthrough here, by design
add e2e test to cover un-toleration while RTE is running, let's ensure the expected behavior

### CODE REVIEW

**Comment by ffromani** [comment]
related to #808

**Review by shajmakh — COMMENTED** [review]
Thanks for this. 2 unblocking notes below
/lgtm

File: test/e2e/serial/tests/tolerations.go [hunk 1]
```diff
@@ -95,14 +98,80 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
...
+
+				defer func(ctx context.Context) {
+					var err error
+					By("resetting tolerations")
+					_ = setRTETolerations(ctx, fxt.Client, nroKey, []corev1.Toleration{})
+					// no need to wait for update - the original change must not have went through
+					By("waiting for DaemonSet to be ready")
+					_, err = wait.With(fxt.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(ctx, dsKey)
+					Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+				}(ctx)
```
[the remark this answers is not in the document]
Comment (ffromani) [reply test/e2e/serial/tests/tolerations.go:161]: good catch, done

File: test/e2e/serial/tests/tolerations.go [hunk 2]
```diff
@@ -95,14 +98,80 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
...
+			})
+
+			It("should handle invalid field: effect [test_id:OCP-72862]", func(ctx context.Context) {
```
[the remark this answers is not in the document]
Comment (ffromani) [reply test/e2e/serial/tests/tolerations.go]: sure, fixed. Thanks for raising!

**Pull request merged**